src/reports.cpp

```diff
@@ -602,31 +604,48 @@ static config unit_moves(reports::context & rc, const unit* u)
 		const terrain_type& info = rc.map().get_terrain_info(terrain);
 
 		if (info.union_type().size() == 1 && info.union_type()[0] == info.number() && info.is_nonnull()) {
+			terrain_moves.push_back(std::make_pair(info.name(), u->movement_cost(terrain)));
+		}
+	}
 
-			const std::string& name = info.name();
-			const int moves = u->movement_cost(terrain);
-
-			tooltip << name << ": ";
-
-			std::string color;
-			//movement  -  range: 1 .. 5, movetype::UNREACHABLE=impassable
-			const bool cannot_move = moves > u->total_movement();
-			if (cannot_move)		// cannot move in this terrain
-				color = "red";
-			else if (moves > 1)
-				color = "yellow";
-			else
-				color = "white";
-			tooltip << "<span foreground=\"" << color << "\">";
-			// A 5 MP margin; if the movement costs go above
-			// the unit's max moves + 5, we replace it with dashes.
-			if(cannot_move && (moves > u->total_movement() + 5)) {
-				tooltip << font::unicode_figure_dash;
-			} else {
-				tooltip << moves;
+	// sort terrain list by movement cost, in same way as in unit_preview_pane.cpp
+	struct
```

Comment:
This function is too short to warrant an entire struct. Could you use a lambda instead?

Reply:
A lambda can work, but I find them way less readable. Considering there shouldn't be a performance difference, I would like to not.

Reply:
I too absolutely prefer the lambda rather than a struct.

Reply:
... I'm not seeing the problem.

Reply:
???

What are you talking about? We're asking you to please use a lambda instead. I was noting that the following type of construct would suffice:

```
auto movement_compare_ascending = [](std::pair<std::string, int> a, std::pair<std::string, int> b) {
    // ...
};
```

Rather than placing the lambda inline in the function call as is typical. I was assuming (perhaps incorrectly) that the latter was your primary reason for finding them unreadable.

As an aside, the arguments to the comparator should probably also be taken as const references, rather than by value.